ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactTextInputManager.java

```diff
@@ -748,6 +748,11 @@ public boolean onEditorAction(TextView v, int actionId, KeyEvent keyEvent) {
                   new ReactTextInputSubmitEditingEvent(
                       editText.getId(),
                       editText.getText().toString()));
+              // if blurOnSubmit is false, and isMultiline, go to new line
```

Comment:
That's not quite correct, I think. It should work this way:
 * blurOnSubmit && isMultiline => generate `submit` event; blur; return true;
 * !blurOnSubmit && isMultiline => return false; // Perform default behaviour.
 * blurOnSubmit && !isMultiline => generate `submit` event; blur; return true;
 * !blurOnSubmit && !isMultiline => return true; // Shallow <enter>